### DISCUSSION TITLE
Copy mesh to sub app when using mortar contact

### DISCUSSION THREAD

**Discussion by lyyc199586** [idaholab/moose#24016] [Q&A Meshing] [answered]
Hi all, I am trying to use mortar contact in the main app and phase field in the sub app. I want to copy the mesh to the sub app and tried the option `clone_parent_mesh=true` but get an error:
```
*** ERROR ***
We should be adding geometric rms so early that we haven't set our MeshBase yet
```

I wonder what is the correct setting to copy mesh to the sub app.
here is part of my input files：
main.i
```
[MultiApps]
  [fracture]
    type = TransientMultiApp
    input_files = fracture.i
    cli_args = 'E=${E};K=${K};G=${G};Lambda=${Lambda};Gc=${Gc};l=${l};sigma_ts=${sigma_ts};sigma_cs=${sigma_cs};delta=${delta}'
    execute_on = 'TIMESTEP_END'
    clone_parent_mesh = true
  []
[]

[Transfers]
  [from_d]
    type = MultiAppCopyTransfer
    from_multi_app = fracture
    variable = 'd'
    source_variable = 'd'
  []
  [to_psie_active]
    type = MultiAppCopyTransfer
    to_multi_app = fracture
    variable = 'disp_x disp_y strain_zz psie_active'
    source_variable = 'disp_x disp_y strain_zz psie_active'
  []
[]

[Mesh]
  coord_type = XYZ
  [fmg]
    type = FileMeshGenerator
    file = '../mesh/disk_mortar_flat.msh'
    ### for recover
    # use_for_exodus_restart = true
    # file = './out/solid_R14.5_ts10_cs80_l0.25_delta25.e'
  []
  patch_size = 10
  patch_update_strategy = always
[]
```

sub.i
```
[Mesh]
  coord_type = XYZ
  [fmg]
    type = FileMeshGenerator
    file = '../mesh/disk_mortar_flat.msh'
    ### for recover
    # use_for_exodus_restart = true
    # file = './out/solid_R14.5_ts10_cs80_l0.25_delta25.e'
  []
  # # patch_update_strategy = auto
[]
```

**Comment by GiudGiud** [comment]
Hello

Just dont use clone_parent_mesh. 
It doesnt save anything, you can just create the mesh in the subapp the same way as the main app

Guillaume

**Reply by lyyc199586** [reply]
should I also use the same `patch_size` and `patch_update_strategy` in sub app?

**Reply by lyyc199586** [reply]
weirdly, even if I use the same mesh setting:
```
[Mesh]
  coord_type = XYZ
  [fmg]
    type = FileMeshGenerator
    file = '../mesh/disk_mortar_flat.msh'
    ### for recover
    # use_for_exodus_restart = true
    # file = './out/solid_R14.5_ts10_cs80_l0.25_delta25.e'
  []
  patch_size = 10
  patch_update_strategy = always
[]
```

I still get an error in copy transfer:
```
The following error occurred in the object "to_psie_active", of type "MultiAppCopyTransfer".

The meshes must be identical to utilize MultiAppDofCopyTransfer::transfer.[39m
```

**Reply by GiudGiud** [reply]
the meshes must be identical. Is the patching changing the mesh differently in either mesh?

**Reply by lyyc199586** [reply]
no I am using the same `patch_size = 10
  patch_update_strategy = always`

**Reply by GiudGiud** [reply, marked as answer]
kk. Just use a shape evaluation transfer. It wont complain

**Reply by GiudGiud** [reply]
THe meshes must be different in some way. I m not familiar with patch_update enough to know why

**Reply by lyyc199586** [reply]
ok, I will try, thanks

**Reply by lindsayad** [reply]
If you are using the `Contact` action, by default you will be adding additional mesh generators from there, unless you set `generate_mortar_mesh = false`. To use that option, you would have to have already added the lower dimensional blocks necessary for mortar yourself
